lz_header.S

```diff
@@ -318,36 +295,33 @@ lz_exit:
 	movl	%ebp, %edx
 	jmp	*%ebx
 
-/* GDT */
 .align 16
-gdt_desc64:
-	.word	gdt_table64_end - gdt_table64 - 1 /* Limit */
-	.quad	lz_offset(gdt_table64) /* Base - dynamically relocated. */
-gdt_desc64_end:
+gdt:
+	/* Null Segment. Reused for 32bit GDTR */
+	.word   0
+gdtr:
+	.word	.Lgdt_end - gdt - 1 /* Limit */
+	.long	lz_offset(gdt)      /* Base - dynamically relocated. */
 
-.align 16
-gdt_table64:
-	/* Null Segment */
-	.quad	0x0000000000000000
-	/* 64b Code Segment */
-	.word	0x0000 /* Limit 1 */
-	.word	0x0000 /* Base 1 */
-	.byte	0x00   /* Base 2 */
-	.byte	0x9a   /* P=1 DPL=0 S=1 Type=1010 C=0 R=1 A=0 */
-	.byte	0x20   /* G=0 D=0 L=1 AVL=0 Limit 2 */
-	.byte	0x00   /* Base 3 */
 	/* 32b Code Segment */
 	.word	0xffff /* Limit 1 */
 	.word	0x0000 /* Base 1 */
 	.byte	0x00   /* Base 2 */
-	.byte	0x9a   /* P=1 DPL=0 S=1 Type=0010 C=0 W=1 A=0 */
+	.byte	0x9b   /* P=1 DPL=0 S=1 Type=0010 C=0 W=1 A=1 */
```

Comment:
They don't need to be marked accessed, although I don't think it impacts code/data segments.

Reply:
It is a performance optimisation.  The processor *will* set the access bit on first use, just like Accessed bits in pagetables.

However, if you're not using the bits for tracking, it is good to set them yourself, which prevents the processor needing to do an atomic operation to set the bit.